addons/web_editor/static/src/js/wysiwyg/widgets/color_palette.js

```diff
@@ -410,10 +453,265 @@ const ColorPaletteWidget = Widget.extend({
             this.trigger_up(eventName, colorInfo);
         }
         this._buildCustomColors();
-        this._markSelectedColor();
         if (this.colorPicker) {
             this.colorPicker.setSelectedColor(colorInfo.color);
         }
+        if (this.gradientColorPicker) {
+            const customGradient = weUtils.isColorGradient(colorInfo.color) ? colorInfo.color : false;
+            this._selectGradient(customGradient);
+        }
+        this._markSelectedColor();
+    },
+    /**
+     * Populates the gradient editor.
+     *
+     * @private
+     * @param {string} gradient CSS string
+     */
+    _selectGradient: function (gradient) {
+        const editor = this.gradientEditorParts;
+        if (this.gradientColorPicker.$el) {
+            this.gradientColorPicker.do_hide();
+        }
+        if (!gradient) {
+            $(editor.customContent).addClass('d-none');
+            editor.customButton.style['background-image'] = '';
+            editor.customButton.dataset.color = false;
+            return;
+        }
+        $(editor.customContent).removeClass('d-none');
+        gradient = gradient.toLowerCase();
+        editor.customButton.style['background-image'] = gradient;
+        editor.customButton.dataset.color = gradient;
+        // The scale display shows the gradient colors horizontally by canceling the type and angle
+        // which are before the first comma.
+        const scaleGradient = gradient.replace(/[^,]+,/, 'linear-gradient(90deg,');
+        editor.scale.style['background-image'] = scaleGradient;
+
+        const isLinear = gradient.startsWith('linear-gradient(');
+        // Extract colors and their positions: colors can either be in the #rrggbb format or in the
+        // rgb/rgba(...) format, positions are are expected to be expressed as percentages.
```

Comment:
- are are
- our editor is not supposed to expect anything but a valid DOM, can position be defined another way? What happens in that case?

Reply:
CSS gradients also accept lengths (100px, 5em...) as positions.
Does the editor need to be able to cope with length mode as well ?
The whole dragging of sliders to determine positions would have to be very different.

Currently, if we select an element with a gradient that uses length positions, we get an error popup because the individual colors and positions are not correctly extracted.

Reply:
Implicitly converting existing lengths to percentage based on the current sizes will be quite complex: relying on the angle, or the radial size settings...

Reply:
We should at least not crash :wink: Maybe something like "disabling" gradient edition (with a gray overlay or something like that) if the given gradient is not understood. Then if the user wants to use our gradient editor he can hit the trash button to remove the gradient and create an Odoo-supported one.

Reply:
What about no considering an "unrecognized gradient" as an existing "custom gradient" ?
- predefined gradients are available
- custom button is black (as if the current color was not a gradient) => upon click, do as if it was a new gradient

Reply:
Would be perfect :+1: